### PR TITLE
feature: Adding Training Compiler support for TensorFlow estimator starting TF 2.9

### PR DESCRIPTION
**Pull request by Lokiiiiii**
*Issue #, if available:*

*Description of changes:*
As title
This is a fast follow up to https://github.com/aws/sagemaker-python-sdk/pull/3154

*Testing done:*
Local testing and PR testing

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I certify that the changes I am introducing will be backword compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/training_compiler/config.py [hunk 1]
```diff
@@ -118,36 +118,25 @@ def _to_hyperparameter_dict(self):
...
-                The `validate` classmethod raises error
-                if an instance type not in the ``SUPPORTED_INSTANCE_CLASS_PREFIXES`` list
-                or ``local`` is passed to the `instance_type` parameter.
-            distribution (dict): A dictionary of the distributed training option that's specified
-                to :class:`~sagemaker.huggingface.HuggingFace`.
-                SageMaker's distributed data parallel and model parallel libraries
-                are currently not compatible
-                with SageMaker Training Compiler.
+            estimator (str): A estimator object
+                If SageMaker Training Compiler is enabled, it will validate whether
```
Comment (mchoi8739) [inline src/sagemaker/training_compiler/config.py:127]:
```suggestion
                When SageMaker Training Compiler is enabled, it validates if
```

File: src/sagemaker/training_compiler/config.py [hunk 2]
```diff
@@ -161,25 +150,28 @@ def validate(
...
                 "SageMaker distributed training configuration is currently not compatible with "
                 "SageMaker Training Compiler."
             )
+
+        if estimator.debugger_hook_config or (not estimator.disable_profiler):
+            helper_string = (
+                "Using Debugger and/or Profiler with SageMaker Training Compiler causes poor "
+                "performance. Found debugger_hook_config={} "
+                "disable_profiler={}. Please set "
+                "debugger_hook_config=None and disable_profiler=True for optimal performance."
```
Comment (mchoi8739) [inline src/sagemaker/training_compiler/config.py:172]:
```suggestion
                "debugger_hook_config=None and disable_profiler=True for optimal performance. For more information, see Training Compiler Performance Considerations (https://docs.aws.amazon.com/sagemaker/latest/dg/training-compiler-tips-pitfalls.html#training-compiler-tips-pitfalls-considerations)."
```

**Comment by mchoi8739** [comment]
Add the following to `doc/frameworks/tensorflow/sagemaker.tensorflow.rst` below the TensorFlow Estimator section and above the TensorFlow Serving Model section. 

```
TensorFlow Training Compiler Configuration
--------------------------------------------

.. autoclass:: sagemaker.tensorflow.TrainingCompilerConfig
    :members:
    :undoc-members:
    :show-inheritance:
```

**Review by mchoi8739 — APPROVED** [review]

File: src/sagemaker/training_compiler/config.py [hunk 3]
```diff
@@ -118,36 +118,25 @@ def _to_hyperparameter_dict(self):
     @classmethod
     def validate(
         cls,
-        image_uri,
-        instance_type,
-        distribution,
+        estimator,
```
Comment (navinsoni) [inline src/sagemaker/training_compiler/config.py:121]: Is this backward compatible as you are removing args from it

File: src/sagemaker/image_uris.py [hunk 4]
```diff
@@ -134,21 +134,18 @@ def retrieve(
...
                 _framework = f"{framework}-{inference_tool}"
         config = _config_for_framework_and_scope(_framework, image_scope, accelerator_type)
-    elif framework == HUGGING_FACE_FRAMEWORK:
-        config = _config_for_framework_and_scope(
-            framework + "-training-compiler", image_scope, accelerator_type
-        )
-    else:
-        raise ValueError(
-            "Unsupported Configuration: Training Compiler is only supported with HuggingFace"
-        )
```
Comment (navinsoni) [inline src/sagemaker/image_uris.py:151]: we need to raise this error right if framework is not "Huggingface

Comment (Lokiiiiii) [reply src/sagemaker/image_uris.py:151]: We now support HF and TF estimators. I don't think there is a need for this anymore.

File: src/sagemaker/training_compiler/config.py (hunk 3 above)
Comment (Lokiiiiii) [reply src/sagemaker/training_compiler/config.py:121]: The validate function is not for direct use by customers. The estimators use it internally to validate customer's args.

To answer your question, this is not backwards compatible.

Comment (navinsoni) [reply src/sagemaker/training_compiler/config.py:121]: Do you anticipate someone might have used it.

Comment (Lokiiiiii) [reply src/sagemaker/training_compiler/config.py:121]: Nope, we have limited customers. 
We have an established process for getting support through our PM.
I have a high level of confidence that nobody is using this function directly.

**Review by navinsoni — APPROVED** [review]

**Review by shreyapandit — APPROVED** [review]

**Pull request merged**